### DISCUSSION TITLE
execute only one activity

### DISCUSSION THREAD

**Discussion by dsoffer** [elsa-workflows/elsa-core#3185] [General]
Is there currently a way to execute one activity of a workflow and return? 

Thanks

Daniel

**Comment by sfmskywalker** [comment]
The way this is usually done is by implementing a [blocking workflow](https://elsa-workflows.github.io/elsa-core/docs/next/guides/guides-blocking-activities), but I have not yet come across a scenario where the application needs precise control over executing one activity and then the next one, effectively circumventing the workflow runner's logic that determines what activity (if any) to run next.

If you share a little bit more about your use case, then perhaps I can point you to the right direction. Or perhaps we need to add support for this in some way.

**Comment by z0ha1b** [comment]
Only one activity is performed in the workflow if it the workflow is manually triggered.

Designer
![image](https://github.com/elsa-workflows/elsa-core/assets/24979642/b503f21b-4d2b-446b-9afa-421afb7e155e)

Handler
![image](https://github.com/elsa-workflows/elsa-core/assets/24979642/858c18a7-c9a4-4518-a4b9-94b82a580038)

GetDocument Activity
![image](https://github.com/elsa-workflows/elsa-core/assets/24979642/9b1cb15f-e4f5-4631-ad7f-1ddd45fa59fc)

Please help me! I want to trigger all the activities with in the workflow that was created in the designer
